experimental/dds/tree2/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -457,18 +462,41 @@ export function applyTypesFromContext(
 			primary !== undefined,
 			0x4d6 /* array data reported comparable with the schema without a primary field */,
 		);
-		const children = applyFieldTypesFromContext(schemaData, primary.schema, data);
+		const children = applyFieldTypesFromContext(context.schemaData, primary.schema, data);
 		return {
 			value: undefined,
 			type,
 			fields: new Map(children.length > 0 ? [[primary.key, children]] : []),
 		};
 	} else {
+		if (
+			context.globalFieldKeySymbol !== undefined &&
+			data[context.globalFieldKeySymbol] === undefined
+		) {
+			const globalFieldSchema = getFieldSchema(
+				context.globalFieldKeySymbol,
+				context.schemaData,
+				schema,
+			) as GlobalFieldSchema;
```

Comment:
"as" casts like this are a [Code smell](https://en.wikipedia.org/wiki/Code_smell). While I can't say if this one is avoidable without deeper investigation the fact that there are a couple of them in here suggests a different design might be better (ex: maybe make work for all fields, or since you know you have a global field, do the lookup in a simpler way thats specific to them (getFieldSchema exists to handle both the global and local field cases, see its implementation for how to handle either on alone)

Reply:
updated to use the `getFieldGenerator` API, and removed code smell